cognite/neat/rules/importers/_inference2rules.py

```diff
@@ -0,0 +1,216 @@
+                           WHERE { ?s a ?class . }
+                           group by ?class order by DESC(?instances)"""
+
+INSTANCES_OF_CLASS_QUERY = """SELECT ?s WHERE { ?s a <class> . }"""
+
+INSTANCE_PROPERTIES_DEFINITION = """SELECT ?property (count(?property) as ?occurrence) ?dataType ?objectType
+                                    WHERE {<instance_id> ?property ?value .
+                                           BIND(datatype(?value) AS ?dataType)
+                                           OPTIONAL {?value rdf:type ?objectType .}}
+                                    GROUP BY ?property ?dataType ?objectType"""
+
+
+class InferenceImporter(BaseImporter):
+    """Rules inference through analysis of knowledge graph provided in various formats.
+
+
+    Args:
+        graph: Knowledge graph
+        max_number_of_instance: Maximum number of instances to be used in inference
+        make_compliant: If True, NEAT will attempt to make the imported rules compliant with CDF
+    """
+
+    def __init__(
+        self, issue_list: IssueList, graph: Graph, max_number_of_instance: int = -1, make_compliant: bool = False
+    ):
+        self.issue_list = issue_list
+        self.graph = graph
+        self.max_number_of_instance = max_number_of_instance
+        self.make_compliant = make_compliant
+
+    @classmethod
+    def from_rdf_file(cls, filepath: Path, make_compliant: bool = False, max_number_of_instance: int = -1):
+        issue_list = IssueList(title=f"'{filepath.name}'")
+
+        graph = Graph()
+        try:
+            graph.parse(filepath)
+        except Exception as e:
+            raise Exception(f"Could not parse RDF file: {e}") from e
+
+        return cls(issue_list, graph, make_compliant=make_compliant, max_number_of_instance=max_number_of_instance)
+
+    @classmethod
+    def from_json_file(cls, filepath: Path, make_compliant: bool = False, max_number_of_instance: int = -1):
+        raise NotImplementedError("JSON file format is not supported yet.")
+
+    @classmethod
+    def from_yaml_file(cls, filepath: Path, make_compliant: bool = False, max_number_of_instance: int = -1):
+        raise NotImplementedError("YAML file format is not supported yet.")
+
+    @classmethod
+    def from_xml_file(cls, filepath: Path, make_compliant: bool = False, max_number_of_instance: int = -1):
+        raise NotImplementedError("JSON file format is not supported yet.")
+
+    @overload
+    def to_rules(self, errors: Literal["raise"], role: RoleTypes | None = None) -> Rules: ...
+
+    @overload
+    def to_rules(
+        self,
+        errors: Literal["continue"] = "continue",
+        role: RoleTypes | None = None,
+    ) -> tuple[Rules | None, IssueList]: ...
+
+    def to_rules(
+        self, errors: Literal["raise", "continue"] = "continue", role: RoleTypes | None = None
+    ) -> tuple[Rules | None, IssueList] | Rules:
+        """
+        Creates `Rules` object from the data for target role.
+        """
+
```

Comment:
As in the `DMSImporter` you will see
```suggestion
    ) -> tuple[Rules | None, IssueList] | Rules:
        """
        Creates `Rules` object from the data for target role.
        """
        
        if self.issue_list.has_errors:
            # In case there were errors during the import, the to_rules method will return None
            return self._return_or_raise(self.issue_list, errors)

```